### PR TITLE
Define application interface

### LINKED ISSUE #8: Application base class

### PR DESCRIPTION
**Pull request by pavelanihimovsky**
resolves #8

### CODE REVIEW

File: aquality_selenium_core/applications/Application.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    @abstractmethod
+    def is_started(self) -> bool:
+        """
+        :return: Is the application already running or not.
+        """
+        pass
+
+    @abstractmethod
+    def set_implicit_wait_timeout(self, value: int) -> None:
```
Comment (mialeska) [inline aquality_selenium_core/applications/Application.py:27]: should we use timedelta for timeouts in Python, like we do with Duration in java and TimeSpan in .NET?

Comment (pavelanihimovsky) [reply aquality_selenium_core/applications/Application.py:27]: Selenium webdriver methos `driver,implicitly_wait` accepts integer. Why do we need to use timedelta ?

Comment (mialeska) [reply aquality_selenium_core/applications/Application.py:27]: it's just selenium a bit frustrating as uses milliseconds as time unit in javascript but seconds in python and ruby...
https://www.selenium.dev/documentation/en/webdriver/waits/
nevermind, just asking

Comment (pavelanihimovsky) [reply aquality_selenium_core/applications/Application.py]: You are right. It'd be better to have timedelta and then get seconds in implementation.

File: aquality_selenium_core/applications/application.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""Defines abstraction of any application (browser, desktop or mobile driver) controlled by Selenium WebDriver API."""
+from abc import ABC
+from abc import abstractmethod
+from datetime import timedelta
+
+from selenium.webdriver import Remote
```
Comment (Nikikuzi) [inline aquality_selenium_core/applications/application.py:6]: Due to type understanding probably will be better to replace to from selenium.webdriver.remote.webdriver import WebDriver

This is the link to same class, but name is not so abstract

File: aquality_selenium_core/applications/application.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+class AbstractApplication(ABC):
+    """Interface of any application controlled by Selenium WebDriver API."""
+
+    @property
+    @abstractmethod
+    def driver(self) -> Remote:
+        """:return: Current instance of driver."""
+        pass
+
+    @abstractmethod
```
Comment (Nikikuzi) [inline aquality_selenium_core/applications/application.py:18]: This is not a method, so let's convert to property

File: requirements/common.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+selenium==3.141.0
```
Comment (Nikikuzi) [inline requirements/common.txt:1]: remove file(dependency already present in setup.cfg, "install_requires" section)

File: requirements/test.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+-r common.txt
```
Comment (Nikikuzi) [inline requirements/test.txt:1]: remove file(dependency added to setup.cfg, "testing" section)

**Pull request merged**